src/Filesystem/Folder.php

```diff
@@ -194,16 +194,32 @@ public function read($sort = true, $exceptions = false, $fullPath = false)
             if ($fullPath) {
                 $name = $item->getPathname();
             }
+
             if ($item->isDir()) {
-                $dirs[] = $name;
+                $dirs[$item->getCTime()][] = $name;
             } else {
-                $files[] = $name;
+                $files[$item->getCTime()][] = $name;
```

Comment:
Ah probably you changed the current behavior. `$item->getCTime()` should be called only if `$sort` is `true`.